### PR TITLE
Fixes to query for 1.8 Update

### PR DESCRIPTION
**Pull request by milleruntime**
This should satisfy ACCUMULO-2446 so we can close it. There is still a bug in query when you search on terms not in the index.  But I will file this as a separate bug since its working otherwise.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -67,7 +68,7 @@
    </dependency>
      <dependency>
       <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-client</artifactId>
```
Comment (madrob) [inline pom.xml]: Can you elaborate on why this change is needed? hadoop-client should be the default entry point for consumers of the APIs

Comment (milleruntime) [reply pom.xml]: I can't remember why I replaced hadoop-client with hadoop-common.  I may have thought there was a class missing that we needed from the common jar.  I just built and tested it again and hadoop-client seems to work just fine.

File: query/src/main/java/org/apache/accumulo/examples/wikisearch/iterator/DefaultIteratorEnvironment.java [hunk 2]
```diff
@@ -18,30 +18,24 @@
 
 import java.io.IOException;
 
-import org.apache.accumulo.core.client.sample.SamplerConfiguration;
+import org.apache.accumulo.core.client.impl.BaseIteratorEnvironment;
```
Comment (joshelser) [inline query/src/main/java/org/apache/accumulo/examples/wikisearch/iterator/DefaultIteratorEnvironment.java]: Bad form to depend on an impl class. There aren't any API guarantees on BaseIteratorEnvironment not changing in a way that would break DefaultIteratorEnvironment. Would be better to leave this as it was (implements IteratorEnvironment).

**Review by joshelser — CHANGES_REQUESTED** [review]
Just the one compatibility issue, otherwise looks fine.

**Comment by joshelser** [comment]
Ship it!
